resources/user-guides/attach-data.md

```diff
@@ -0,0 +1,113 @@
+# Using the Attach Data Rules Action
+
+The Attach Data rule action was added in the `Mobile Core v2.1.8` Launch extension. Attach Data is powerful and complex, and is recommended for users with advanced use cases.
+
+In order to understand how to use Attach Data, you will need a foundational knowledge on how Events flow in the SDK and the role of the SDK's Rules Engine.
+
+This document provides the information necessary to understand how to use Attach Data, and provides an example of how to use it.
+
+## SDK Events
+
+#### What is an Event?
+
+In the SDK, Events hold all of the data required by other SDK extensions to perform their necessary actions.
+
+Events have the following defining properties:
+
+- `type` - describes what kind of Event this is. _E.g._ **Analytics**, **Target**, or **Lifecycle**.
+- `source` - indicates the cause of the Event, primarily used to indicate the directionality of the Event. _E.g._ **request** or **response**.
+- `event data` - any additional data needed to fully define the Event. _E.g._ context data on an Analytics event.
+
+Extensions that register with the SDK will also register Event listeners. A listener is defined by an Event `type` & `source` combination. When the `event hub` processes an Event, it notifies all listeners that match its combination.
+
+#### When are Events created in the SDK
+
+Events are created by an extension then dispatched to the `event hub` in the SDK's core. Each of the Rules created in Adobe Launch are then evaluated against the current Event. Finally, the Event is passed along to each of the listeners for Events with this `type/source` combination.
+
+{% hint style="important" %}
+Invoking any of the SDK's public APIs will result in an Event being created and dispatched. Attach Data use cases tend to focus on these types of Events.
```

Comment:
Please reword the note as follows:
"When you Invoke any of the SDK's public APIs, an Event is created and dispatched. Attach Data use cases usually focus on these types of Events.